addon/background-script.js

```diff
@@ -0,0 +1,13 @@
+let ad = document.querySelectorAll('iframe')
+
+for (let i = 0; i < ad.length; i++ ){
+  let father = ad[i].parentElement
+  father.removeChild(ad[i])
+
+const img = document.createElement('img')
+
+  for (let z = 0; z < images.length; z++ ){
```

Comment:
what are u trying to do with this second for loop? there is no variable called `images` so `images.length` is undefined, which means this loop won't actually do anything (other than potentially cause a bug), unless i'm misunderstanding something u're trying to do (let me know if that's the case) i think u should remove this (as well as it's closing `}` on line 12. the 2 lines inside this loop.

Reply:
I intended to use this loop to put different images from the array of uploaded images into the places where the ads were.

Reply:
i see. the problem is that array doesn't exist, u need to request it from ur server's API using the `fetch()` function (that's how u request data from APIs) i'll explain in more detail below.